Setup minimum dependency testing

bhazelton opened the pull request:
Also some minor dependency cleanup in setup.py and test environment yaml.

aelanman commented on pyradiosky/skymodel.py:
```diff
@@ -399,6 +398,8 @@ def healpix_to_sky(hpmap, indices, freqs):
     -----
     Currently, this function only converts a HEALPix map with a frequency axis.
     """
+    import astropy_healpix
```
Would it be better to have this fail with a message like, "astropy-healpix module required to use HEALPix methods"?

bhazelton replied:
yes, I'll add that.

bhazelton commented:
@aelanman I think this is ready for you again. I addressed your comment, and did a little other cleanup and then rebased it after Ruby's PR was merged in.
